File: Angular+.NET+MSSQL/EverythingFormApp/EverythingFormAPI/EverythingFormAPI/Models/EverythingFormModel.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace EverythingFormAPI.Models
{
    public class EverythingFormModel
    {
        //public int? Id { get; set; }

        [Required(ErrorMessage = "Name is required")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Age is required")]
        [Range(18, 100, ErrorMessage = "Age must be between 18 and 100")]
        public int Age { get; set; }

        [Required(ErrorMessage = "Gender is required")]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public GenderType? Gender { get; set; }

        [Required(ErrorMessage = "Email is required")]
        [EmailAddress(ErrorMessage = "Invalid email address"    )]
        public string Email { get; set; }

        [Required(ErrorMessage = "Password is required")]
        [MinLength(8, ErrorMessage = "Password must be at least 8 characters long")]
        public string Password { get; set; }

        [Required(ErrorMessage = "SelectSingle is required")]
        public int SelectSingle { get; set; }

        [Required(ErrorMessage = "SelectMultiple is required")]
        public int[] SelectMultiple { get; set; }

        [Required(ErrorMessage = "Textarea is required")]
        public string Textarea { get; set; }

        // Additional properties can be added as needed
        public IFormFile? FileInput { get; set; }
        public enum GenderType
        {
            Male,
            Female
        }
    }
}

File: Angular+.NET+MSSQL/EverythingFormApp/EverythingFormUI/src/app/models/everything-form.model.ts
export interface EverythingForm {
    id?: number | null; // Optional as the ID will be assigned by the server
    name: string;
    age: number;
    gender: GenderType;
    email: string;
    password: string;
    selectSingle: number;
    selectMultiple: number[];
    textarea: string;
    fileInput?: File | null; // Optional as file input is not always present
    checkbox: boolean;
}
export enum GenderType {
    Male = 'Male',
    Female = 'Female'
}